packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts

```diff
@@ -458,6 +444,35 @@ export class TaskDefinition extends TaskDefinitionBase {
       throw new Error('Cannot specify runtimePlatform in non-Fargate compatible tasks');
     }
 
+    //FARGATE compatible tasks pre-checks
+    if (this.isFargateCompatible) {
+      if (this.networkMode !== NetworkMode.AWS_VPC) {
+        throw new Error(`Fargate tasks can only have AwsVpc network mode, got: ${this.networkMode}`);
+      }
+
+      if (props.placementConstraints && props.placementConstraints.length > 0) {
+        throw new Error('Cannot set placement constraints on tasks that run on Fargate');
+      }
+
+      if (!props.cpu || !props.memoryMiB) {
+        throw new Error(`Fargate-compatible tasks require both CPU (${props.cpu}) and memory (${props.memoryMiB}) specifications`);
+      }
+
+      if (props.inferenceAccelerators && props.inferenceAccelerators.length > 0) {
+        throw new Error('Cannot use inference accelerators on tasks that run on Fargate');
+      }
+
+      // Validate the CPU and memory size for the Windows operation system family.
+      if (props.runtimePlatform?.operatingSystemFamily?.isWindows()) {
+        this.checkFargateWindowsBasedTasksSize(props.cpu, props.memoryMiB, props.runtimePlatform);
+      }
```

Comment:
Might've missed this before, but this check was moved inside the `(this.isFargateCompatible)` block, when before it was outside of it. Just wondering if this check is only supposed to happen for Fargate-related things and if it's safe to have moved behind this check.

Reply:
If the task def does not have FARGATE capability then task level CPU / mem does not apply for windows containers. 
![Screenshot 2024-07-30 at 5 22 30 PM](https://github.com/user-attachments/assets/5b847929-964a-49ca-af4b-60392c409a87)
Plus, 
> For task definitions that specify FARGATE for the requiresCompatibilities parameter (even if EC2 is also specified), you must use one of the values in the following table. These values determines your range of supported values for the CPU and memory parameter.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html

Hence the if condition would be good

Reply:
Can we move it back outside of the `(this.isFargateCompatible)` check anyway? Adding the additional logic is still technically changing the behaviour of the class, and there's a risk that it might affect something further downstream.

Reply:
@Leo10Gama Moved it outside the fargate checks